client/scripts/components/Message/ComposeWithMenus.js

```diff
@@ -83,6 +84,11 @@ class ComposeWithMenu extends Component {
         : nametags.map(n => n.name)
     }
 
+    this.getTypingPrompt = (nametagId) => {
```

Comment:
I think it would make more sense to, instead of doing this lookup, build a map in the connected component's `mapStateToProps`.

Reply:
Great idea.